library/components/TTooltip.vue

```diff
@@ -1,28 +1,32 @@
 <template>
-  <div
-    @mouseover="handleMouseOver"
-    @mouseleave="handleMouseLeave"
-    ref="tooltipContainer"
-  >
-    <div ref="trigger">
+  <div ref="tooltipContainer">
+    <div
+      ref="trigger"
+      @mouseleave="handleMouseLeave"
+      @mouseover="handleMouseOver"
+    >
       <slot name="trigger">
         <help-circle-outline-icon :size="12" />
       </slot>
     </div>
 
     <!-- $slots.default is used in cases when the trigger is using v-if on content's div -->
     <transition name="fade">
-      <div
-        class="p-2 bg-[#1C1C21] text-white text-sm w-max h-max rounded fixed z-50"
-        :style="{ width, ...positions.tooltip }"
-        ref="tooltip"
-        v-show="isVisible && $slots.default"
-      >
-        <slot></slot>
+      <div class="relative w-0 h-0" :style="{ ...positions.tooltip }">
```

Comment:
Change 1:
Having a parent div that has no height or width but relative position will make sure we do not have keep the exact position inside DOM for tooltip placement, it always stays near trigger.